projects/mci/test/test_drive_fsm.c

```diff
@@ -0,0 +1,151 @@
+#include "can.h"
+#include "can_ack.h"
+#include "can_msg_defs.h"
+#include "can_transmit.h"
+#include "delay.h"
+#include "event_queue.h"
+#include "exported_enums.h"
+#include "gpio.h"
+#include "interrupt.h"
+#include "log.h"
+#include "ms_test_helpers.h"
+#include "soft_timer.h"
+#include "test_helpers.h"
+
+#include "drive_fsm.h"
+#include "mci_events.h"
+#include "motor_controller.h"
+
+static CanStorage s_can_storage;
+static MotorControllerStorage s_mci_storage;
+
+void prv_setup_system_can() {
+  CanSettings can_settings = {
+    .device_id = SYSTEM_CAN_DEVICE_MOTOR_CONTROLLER,
+    .bitrate = CAN_HW_BITRATE_500KBPS,
+    .rx_event = MCI_CAN_EVENT_RX,
+    .tx_event = MCI_CAN_EVENT_TX,
+    .fault_event = MCI_CAN_EVENT_FAULT,
+    .tx = { GPIO_PORT_A, 12 },
+    .rx = { GPIO_PORT_A, 11 },
+    .loopback = true,
+  };
+
+  can_init(&s_can_storage, &can_settings);
+}
+
+void prv_mci_storage_init(void *context) {
+  MotorControllerStorage *storage = context;
+  PrechargeStorage precharge_storage = { .precharge_control = { .port = GPIO_PORT_A, .pin = 9 },
```

Comment:
you're manually filling out `precharge_storage` and that's pretty sketchy because if you change the internals of your `precharge_monitor` module, this test will break for no reason. Merge this module with `motor_controller` and have the init function initialize the `prechare_monitor` module. Alternatively, you can have `precharge_monitor` expose precharge status as a global, and just read it here... that's actually better. 
My description is a lil sloppy, if you didn't understand it ask me.

Reply:
you fixed this lol